backend/LexBoxApi/Controllers/ProjectController.cs

```diff
@@ -187,29 +189,44 @@ public record HgCommandResponse(string Response);
     [ProducesResponseType(StatusCodes.Status200OK)]
     [ProducesResponseType(StatusCodes.Status404NotFound)]
     [ProducesDefaultResponseType]
-    public async Task<ActionResult<HgCommandResponse>> HgVerify(string code)
+    public async Task HgVerify(string code)
     {
         var migrationStatus = await lexBoxDbContext.Projects.Where(p => p.Code == code)
             .Select(p => p.MigrationStatus)
             .FirstOrDefaultAsync();
-        if (migrationStatus is not ProjectMigrationStatus.Migrated) return NotFound();
+        if (migrationStatus is not ProjectMigrationStatus.Migrated)
+        {
+            // Used to return NotFound() but now we have to write the response manually
+            Response.StatusCode = 404;
+            await Response.CompleteAsync();
+            return;
+        }
         var result = await hgService.VerifyRepo(code, HttpContext.RequestAborted);
-        return new HgCommandResponse(result);
+        var writer = Response.BodyWriter;
+        Response.ContentType = "text/plain; charset=utf-8";
+        await result.CopyToAsync(writer.AsStream());
     }
 
     [HttpGet("hgRecover/{code}")]
     [AdminRequired]
     [ProducesResponseType(StatusCodes.Status200OK)]
     [ProducesResponseType(StatusCodes.Status404NotFound)]
     [ProducesDefaultResponseType]
-    public async Task<ActionResult<HgCommandResponse>> HgRecover(string code)
+    public async Task HgRecover(string code)
     {
         var migrationStatus = await lexBoxDbContext.Projects.Where(p => p.Code == code)
             .Select(p => p.MigrationStatus)
             .FirstOrDefaultAsync();
-        if (migrationStatus is not ProjectMigrationStatus.Migrated) return NotFound();
+        if (migrationStatus is not ProjectMigrationStatus.Migrated)
+        {
+            // Used to return NotFound() but now we have to write the response manually
+            Response.StatusCode = 404;
+            await Response.CompleteAsync();
+            return;
+        }
         var result = await hgService.ExecuteHgRecover(code, HttpContext.RequestAborted);
-        return new HgCommandResponse(result);
+        var writer = Response.BodyWriter;
+        await result.CopyToAsync(writer.AsStream());
```

Comment:
content type missing. I'd probably make a method to call from both recover and verify so you don't need to keep them in sync

Reply:
Sure.

Reply:
Done in commit c9471861.